Добавление ТЗ 8

booogiman opened the pull request:
Модели данных,Контроллеров,валидации,логирования,добавление дополнительной валидации анотациями

booogiman committed:
Добавление
Модели данных,Контроллеров,валидации,логирования,добавление дополнительной валидации анотациями

eugene-lenkevich commented on src/main/java/ru/yandex/practicum/filmorate/exceptions/ValidationException.java:
```diff
@@ -0,0 +1,7 @@
+package ru.yandex.practicum.filmorate.exceptions;
+
+public class ValidationException extends RuntimeException {
```
Пользователь ввел неправильные данные - более чем штатная, предсказуемая, ситуация. Здесь стоит использовать проверяемое (checked) исключение: extends Exception(ну и обработать его выше по стеку, на то оно и проверяемое =)

eugene-lenkevich commented on src/main/java/ru/yandex/practicum/filmorate/model/FilmController.java:
```diff
@@ -0,0 +1,66 @@
+
+@RestController
+public class FilmController {
+    private static final Logger log = LoggerFactory.getLogger(FilmController.class);
```
Логгер не обязательно прописывать руками, можно добавить аннотацию к классу @Slf4j библиотеки lombok, например :)

eugene-lenkevich commented on src/main/java/ru/yandex/practicum/filmorate/model/FilmController.java:
```diff
@@ -0,0 +1,66 @@
+    @PostMapping(value = "/films")
+    public void create(@Valid @RequestBody Film film) {
+
+        if (film.getName().isBlank() || film.getName() == null) {
```
Тут можно использовать 
StringUtils.hasText
или же аналогичную функцию из apache commons 
StringUtils.isNotBlank
=)

eugene-lenkevich reviewed (changes requested):
Привет, Лев, спасибо за PR

Я отметил несколько небольших моментов, критичен только момент про исключения
Ну и проверь, пожалуйста, проходят ли  автоматические тесты :)

booogiman committed:
Исправления
Изменения
FilmController.java и типа exсeption
Добавление
Аннатации логирования.

eugene-lenkevich reviewed (approved):
Все супер, спасибо!
